### PR TITLE
Error converter rework

### PR DESCRIPTION
**Pull request by unpluggedk**
Move map information to a separate file, making it easier to broker to use.

### CODE REVIEW

File: ADAL/src/ADAuthenticationErrorMap.m [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                                   @(MSIDErrorAuthorizationFailed): @(AD_ERROR_SERVER_AUTHORIZATION_CODE)
+                                   },
+                           ADHTTPErrorCodeDomain: @{
+                                   @(MSIDErrorServerUnhandledResponse) : @(AD_ERROR_UNEXPECTED)
+                                   }
+                           };
+    
+    s_userInfoKeyMapping = @{
+                             MSIDHTTPHeadersKey : ADHTTPHeadersKey,
+                             MSIDOAuthSubErrorKey : ADSuberrorKey
```
Comment (oldalton) [inline ADAL/src/ADAuthenticationErrorMap.m]: There was one more key added in dev, make sure you apply latest changes (I think it was user displayable id)

**Comment by unpluggedk** [comment]
Needs error map checking.

**Comment by oldalton** [comment]
Same was done here :) https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/1374

**Comment by unpluggedk** [comment]
Unfortunately, not everything was mapped :/ working on it now!

**Pull request merged**